File: java/grains/src/main/java/Grains.java
import java.math.BigInteger;

class Grains {

    BigInteger grainsOnSquare(final int square) {

        if (square < 1 || square > 64) {
            throw new IllegalArgumentException("square must be between 1 and 64");
        }

        BigInteger result = BigInteger.ONE;
        for (int i = 1; i < square; i++) {
            result = result.multiply(BigInteger.TWO);
        }
        return result;
    }

    BigInteger grainsOnBoard() {
        BigInteger result = BigInteger.ONE;
        BigInteger sum = BigInteger.ONE;
        for (int i = 1; i < 64; i++) {
            result = result.multiply(BigInteger.TWO);
            sum = sum.add(result);
        }
        return sum;
    }

}

File: fsharp/grains/Grains.fs
module Grains

let square (n: int): Result<uint64,string> =
    if n > 0 && n <= 64 then
        Ok (pown 2UL (n - 1))
     else
        Error "square must be between 1 and 64"


let total: Result<uint64,string> =
    [ 1 .. 64 ]
    |> List.sumBy (fun num -> pown 2UL (num - 1)) |> Ok
